src/textual/css/stylesheet.py

```diff
@@ -297,13 +297,16 @@ def apply(self, node: DOMNode, animate: bool = False) -> None:
 
         # For each rule declared for this node, keep only the most specific one
         get_first_item = itemgetter(0)
-        node_rules: RulesMap = cast(
-            RulesMap,
-            {
-                name: max(specificity_rules, key=get_first_item)[1]
-                for name, specificity_rules in rule_attributes.items()
-            },
-        )
+        node_rules: RulesMap = cast(RulesMap, {})
```

Comment:
Humor me. I suspect this is more work that required.

Can you restore this code, and change L291 to:

```python
    for rule in reversed(self.rules):
```

Reply:
That works, good spot!